C++/source/number_theory/Median_In_a_Stream.cpp

```diff
@@ -0,0 +1,77 @@
+/*
+    Median in a stream of integers(running integers)
+    ------------------------------------------------
+    An algorithm to find the median of elements read so far from
+    a data stream in a efficient way.
+
+    Assumptions: There are no duplicates.
+
+    Time Complexity
+    ---------------
+    O(NlogN), where N is the number of elements.
+
+    Space Complexity
+    ----------------
+    O(N^2)
+
+    Auxilliary Space
+    ----------------
+    O(N)
+
+*/
+//Explanation link :https://www.youtube.com/watch?v=VmogG01IjYc
+#include<iostream>
+#include<algorithm>
+#include<set>
+#include<math.h>
+#include<iterator>
+#include<queue>
+
+using namespace std;
+
+typedef long long ll;
+priority_queue <ll> great;
+priority_queue<ll,vector <ll>,greater<ll> > shorter;
+
+void add(ll num)
+{
+	if(!great.empty() and num < great.top())
+	{
+		great.push(num);
+		if (great.size() - shorter.size() == 1)
+		{
+			shorter.push(great.top());
+			great.pop();
+		}
+	}
+	else
+	{
+		shorter.push(num);
+		if (shorter.size() - great.size() == 2)
+		{
+			great.push(shorter.top());
+			shorter.pop();
+		}
+	}
+}
+ll median()
+{
+	if (shorter.size() == great.size())
+		return (shorter.top() + great.top()) / 2.0;
+	return shorter.top();
+}
+int main()
+{
+	ios::sync_with_stdio(false);
+	cin.tie(NULL);
+	ll t;
+	cin>>t;
+	while(t--)
```

Comment:
Give a better name for `t`, such as `num_elements`. Also, print a prompt for accepting the input from the user. Eg.
> Enter the number of elements : 

On another note, it would be better to not ask for the number of integers, but instead have a way of determining when to stop reading them. This could be done by using some string (eg. `---`) as the end marker, but for this the input should be taken as a string, checked if it is not the end marker, and then converted to an integer to calculate the median.